datadog.go

```diff
@@ -0,0 +1,137 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// Package datadog contains a Datadog exporter.
+package datadog
+
+import (
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+const (
+	defaultHost   = "localhost"
+	defaultPort   = "8125"
+	opencensusTag = "source:Opencensus"
+)
+
+var (
+	exporter *Exporter
+	tags     = []string{opencensusTag}
+	reg      = regexp.MustCompile("[^a-zA-Z0-9]+")
+)
+
+// Exporter exports stats to Datadog.
+type Exporter struct {
+	opts      Options
+	collector *collector
+}
+
+// ExportView exports to Datadog if view data has one or more rows.
+func (e *Exporter) ExportView(vd *view.Data) {
+	if len(vd.Rows) == 0 {
+		return
+	}
+	e.collector.addViewData(vd)
+}
+
+// Options contains options for configuring the exporter.
+type Options struct {
+	// Namespace specifies the namespace to which metrics are appended.
+	Namespace string
+
+	// Host for DogStatsD connection
+	Host string
+
+	// Port for DogStatsD connection
+	Port string
+
+	// OnError will be called in the case of an error while uploading the stats.
+	// If not set, errors are simply logged.
+	OnError func(err error)
+
+	// Tags specifies a set of global tags to attach to each metric.
+	Tags []string
+}
+
+// getEndpoint determines the address for the statsD client
+func (o *Options) getEndpoint() string {
+	host, port := o.Host, o.Port
+	if host == "" || port == "" {
+		host, port = defaultHost, defaultPort
```

Comment:
I'm hoping you'll follow my other comment and remove this function, but if you don't, beware that this won't work as expected if you have a `host` but not `port`. The `host` will be overridden with the default. What you want here is:
```go
if host == "" {
    host = defaultHost
}
if port == "" {
    port = defaultPort
}
```
Another reason why it's better to stick to the standard "host:port" instead of us going through this trouble.